src/sqlfluff/dialects/dialect_redshift.py

```diff
@@ -317,7 +317,7 @@ class CreateUserSegment(BaseSegment):
     match_grammar = Sequence(
         "CREATE",
         "USER",
-        Ref("NakedIdentifierSegment"),
+        Ref("ObjectReferenceSegment"),
```

Comment:
Agree with these changes but you should add some additional unit tests where these segments are quoted.